docs/security-assessment.md

```diff
@@ -0,0 +1,99 @@
+
+| V004 | Lack of encryption for data in transit, that may expose sensitive data to the risk of interception and compromise.	 |
+| ------------------------- | ------------------------- |
+| Element | PURIS Product |
+| Before Mitigation | Impact: High, Likelihood: High, Risk: High |
+| After Mitigation | Impact: Low, Likelihood: Low, Risk: Low |
+| Mitigation | Product Team currently working on SSL. Team already implemented and enforce HTTPS for  for the front end of the product.Additionally tom is able to serve the backend with spring configuration respectively.Guide was created on how to configure HTTPS  with docker. There are currently some Cores issues that were detected and are being investigated, probably needs spring reconfiguration. It was fixed. Issues happened cause of self signed certificates. The issue is already solved. SSL was also already integrated to the Product. |
+
+| V005 | Lack of rate limiting on API level, that make API vulnerable for denial of service	 |
+| ------------------------- | ------------------------- |
+| Element | PURIS Product |
+| Before Mitigation | Impact: Medium, Likelihood: Medium, Risk: Medium |
+| After Mitigation | Impact: Low, Likelihood: Low, Risk: Low |
+| Mitigation | Implementation was currently postponed till all of the other High findings will be addressed. |
+
+| V006 | Lack of logging and monitoring solution in place, that can hinder the detection of security incidents, performance issues and operational anomalies. |
+| ------------------------- | ------------------------- |
+| Element | PURIS Product |
+| Before Mitigation | Impact: Low, Likelihood: Medium, Risk: Medium |
+| After Mitigation | Impact: Low, Likelihood: Low, Risk: Low |
+| Mitigation | Application is already logging every information about : every call to the respective EDC's, actions related to the exchange of data between partners, all authentication & authorization data Logging enchantments were completed. |
+
+| V007 | Encryption of confidential data at rest. |
+| ------------------------- | ------------------------- |
+| Element | PURIS Product |
+| Before Mitigation | Impact: High, Likelihood: Low, Risk: Medium |
+| After Mitigation | Impact: Low, Likelihood: Low, Risk: Low |
+| Mitigation |  Will be addressed with lower priority due to severity. Goal is to show app is capable of processing the encrypted data. Remaining work is focused on testing those functionality with PostgreSQL DB. |
+
+| V008 | Confirmed vulnerabilities with high severity for H2 Database. |
+| ------------------------- | ------------------------- |
+| Element | H2 DB |
+| Before Mitigation | Impact: High, Likelihood: Medium, Risk: High |
+| After Mitigation | Impact: Low, Likelihood: Low, Risk: Low |
+| Mitigation | PostgreSQL DB has been implemented to the product. Status of implementation is already completed. |
```

Comment:
Actually this wasn't H2 but Hyper SQL DB (hsqldb) -> I initially was wrong about that.

Additionally: We don't include it in compile, but in test scope so that we have easy testing but security when deploying. Should we note that here?

Reply:
I added your comment in the mitigation. Thanks!